base/array.jl

```diff
@@ -64,6 +64,12 @@ const DenseVector{T} = DenseArray{T,1}
 const DenseMatrix{T} = DenseArray{T,2}
 const DenseVecOrMat{T} = Union{DenseVector{T}, DenseMatrix{T}}
 
+## More constructors
+Array{T,N}(inds::NTuple{N,<:OneTo}) where {T,N} = Array{T,N}(length.(inds))
+Array{T}(inds::NTuple{N,<:OneTo}) where {T,N} = Array{T,N}(inds)
+Array{T,N}(ind::OneTo, inds::Vararg{<:OneTo}) where {T,N} = Array{T,N}((ind, inds...))
+Array{T}(ind::OneTo, inds::Vararg{<:OneTo}) where T = Array{T}((ind, inds...))
```

Comment:
We probably want to move away from this kind of array constructor.

Reply:
OK. Some other array types that use unconventional indices can be constructed with `OffsetArray(A, -3:3, 0:5)` so I thought it wouldn't hurt to have something similar here. Will drop.